tuxemon/item/effects/evolve.py

```diff
@@ -42,10 +43,15 @@ class EvolveEffect(ItemEffect):
     monster_evolve: str
 
     def apply(self, target: Monster) -> EvolveEffectResult:
-        monster_evolve = self.monster_evolve
-
-        if any(d["monster_slug"] == monster_evolve for d in target.evolutions):
-            self.user.evolve_monster(target, monster_evolve)
+        if self.monster_evolve == "random":
+            choices = target.evolutions
+            evolution = random.choice(choices).monster_slug
+            self.user.evolve_monster(target, evolution)
             return {"success": True}
         else:
-            return {"success": False}
+            for evolution in target.evolutions:
+                if evolution.monster_slug == self.monster_evolve:
+                    self.user.evolve_monster(target, self.monster_evolve)
+                    return {"success": True}
+                else:
+                    return {"success": False}
```

Comment:
Evolution will only work if it's the first evolution option.

Reply:
Ok, I got it:
if monster A has
evolution1 standard
evolution2_ item
evolution3_ item

Reply:
Do I need to put the pop-up yes/no? In this way I can address through instance_id

Reply:
You have a for loop but you always exit after the first entry.

You should move the success false statement out of the for loop.